curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6448d62ce222044458b75931.md

```diff
@@ -0,0 +1,178 @@
+---
+id: 6448d62ce222044458b75931
+title: Step 5
+challengeType: 0
+dashedName: step-5
+---
+
+# --description--
+
+Your `Convert` button should be working now. But it could get tiring for users to enter in a number, then click that button each time they want to convert from decimal to binary. It would be much more convenient to perform the conversion when the `Enter` or `Return` key is pressed.
+
+The `keydown` event fires every time a user presses a key on their keyboard, and is a good way to add more interactivity to `input` elements.
+
+Chain `.addEventListener()` to `numberInput`. The event listener should listen for `keydown` events and take an empty arrow function as a callback.
+
+# --hints--
+
+You should call the `.addEventListener()` method on your `numberInput` element.
+
+```js
+assert.match(code, /numberInput\s*\.\s*addEventListener\s*\(/);
+```
+
+Your `.addEventListener()` method should listen for the `keydown` event.
+
+```js
+assert.match(code, /numberInput\s*\.\s*addEventListener\s*\(\s*('|"|`)keydown\1\s*,/);
```

Comment:
commas should be optional because the camper technically followed instructions with this code here `numberInput.addEventListener('keydown') `

```suggestion
assert.match(code, /numberInput\s*\.\s*addEventListener\s*\(\s*('|"|`)keydown\1\s*,?/);
```